src/features/game/expansion/components/resources/components/ChestReward.tsx

```diff
@@ -0,0 +1,136 @@
+import { Panel } from "components/ui/Panel";
+import React, { useContext, useEffect, useRef, useState } from "react";
+import { Modal } from "react-bootstrap";
+
+import { Reward } from "features/game/types/game";
+
+import { Button } from "components/ui/Button";
+import { ITEM_DETAILS } from "features/game/types/images";
+import { Context } from "features/game/GameProvider";
+
+import token from "assets/icons/token_2.png";
+import { StopTheGoblins } from "features/farming/crops/components/StopTheGoblins";
+import { ChestCaptcha } from "features/farming/crops/components/ChestCaptcha";
+
+interface Props {
+  reward: Reward | null;
+  rewardType: string;
+  index: number;
+  expansionIndex: number;
+  onCollected: (success: boolean) => void;
+}
+
+type Challenge = "goblins" | "chest";
+
+export const ChestReward: React.FC<Props> = ({
+  reward,
+  rewardType,
+  onCollected,
+  expansionIndex,
+  index,
+}) => {
+  const { gameService } = useContext(Context);
+  const [opened, setOpened] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const challenge = useRef<Challenge>(
+    Math.random() > 0.3 ? "chest" : "goblins"
+  );
+
+  useEffect(() => {
+    if (reward) {
+      setLoading(true);
+      setTimeout(() => setLoading(false), 750);
+    }
+  }, [reward]);
+
+  if (!reward) {
+    return null;
+  }
+
+  const open = () => {
+    setOpened(true);
+    if (rewardType === "Tree") {
+      gameService.send("treeReward.collected", {
```

Comment:
I think we can make this component even dumber by passing the action as a prop:

```
onOpen: () => void;

<ChestReward onOpen={() => gameService.send("treeReward.collected", {         
        index,
        expansionIndex,
 })}/>
```

Likewise for the other situations like crop rewards we will just pass this in as a prop. 

Then we can just call the prop in in the `open` functions

```
const open = () => {
   setIsOpen(true)
   onOpen()
}
```

Reply:
👀👀👀😳 You just blowed my mind

Working on it